src/bin/lua/scopedenum.lua

```diff
@@ -0,0 +1,144 @@
+-- tolua: enumeration class
+-- Written by Waldemar Celes
+-- TeCGraf/PUC-Rio
+-- Jul 1998
+-- $Id: enumerate.lua,v 1.3 2000/01/24 20:41:15 celes Exp $
+
+-- This code is free software; you can redistribute it and/or modify it.
+-- The software provided hereunder is on an "as is" basis, and
+-- the author has no obligation to provide maintenance, support, updates,
+-- enhancements, or modifications.
```

Comment:
Well it was all just copied from `enumerate.lua` with some minor tweaks.  I haven't looked at the license though.